### PR TITLE
Fix small bug that raised error under some conditions

### LINKED ISSUE #11800: IndexError raised when plotting ICA sources using evoked data
### Description of the problem

Using ICA.plot_sources raises an IndexError under specific circumstances. The circumstances are:

-- User is plotting using evoked input
-- User requests to plot a selection of ICA components with gaps (e.g. [0, 2] does not work, but [0, 1, 2] does) 

### Steps to reproduce

```Python
import os
import mne
from mne.preprocessing import ICA

sample_dir = mne.datasets.sample.data_path()
sample_raw = os.path.join(sample_dir, 'MEG', 'sample', 'sample_audvis_raw.fif')
raw = mne.io.read_raw_fif(sample_raw)

raw.crop(tmax=40)
raw.pick_types(meg='mag')
raw.load_data()

ica = ICA(n_components=10)
ica.fit(raw)

epochs = mne.make_fixed_length_epochs(raw)
evoked = epochs.average()

ica.plot_sources(epochs, picks=0)       # Works fine
ica.plot_sources(epochs, picks=[0, 1])  # Works fine
ica.plot_sources(epochs, picks=1)       # Works fine

ica.plot_sources(evoked, picks=0)       # Works fine
ica.plot_sources(evoked, picks=[0, 1])  # Works fine
ica.plot_sources(evoked, picks=1)       # IndexError
```

### Link to data

_No response_

### Expected results

No index error and the corresponding plot.

### Actual results
```
Traceback (most recent call last):
  File "K:\Code\SR-switches-analysis-meg\scratch.py", line 25, in <module>
    ica.plot_sources(evoked, picks=1)       # IndexError
  File "C:\Users\Joshua\anaconda3\envs\mne2023\lib\site-packages\mne\preprocessing\ica.py", line 2226, in plot_sources
    return plot_ica_sources(self, inst=inst, picks=picks,
  File "C:\Users\Joshua\anaconda3\envs\mne2023\lib\site-packages\mne\viz\ica.py", line 112, in plot_ica_sources
    fig = _plot_ica_sources_evoked(
  File "C:\Users\Joshua\anaconda3\envs\mne2023\lib\site-packages\mne\viz\ica.py", line 689, in _plot_ica_sources_evoked
    color, style = label_props[ii]
IndexError: list index out of range
```

### Additional information

```
mne:              1.3.1
numpy:            1.23.5 {MKL 2022.1-Product with 4 threads}
scipy:            1.10.1
matplotlib:       3.7.1 {backend=QtAgg}

sklearn:          1.2.1
numba:            0.56.4
nibabel:          5.0.1
nilearn:          0.10.0
dipy:             1.6.0
openmeeg:         2.5.5
cupy:             Not found
pandas:           1.5.3
pyvista:          0.38.3 {OpenGL 4.5.0 - Build 30.0.101.1692 via Intel(R) UHD Graphics 620}
pyvistaqt:        0.9.1
ipyvtklink:       0.2.2
vtk:              9.2.6
qtpy:             2.3.0 {PyQt5=5.15.6}
ipympl:           Not found
pyqtgraph:        0.13.2
pooch:            v1.7.0

mne_bids:         Not found
mne_nirs:         Not found
mne_features:     Not found
mne_qt_browser:   0.4.0
mne_connectivity: Not found
mne_icalabel:     Not found
```

### PR DESCRIPTION
**Pull request by jCalderTravis**
Fixes #11800.

#### What does this implement/fix?
ICA.plot_sources raised an indexing error under specific conditions described in #11800. This error is no longer raised.

The problem was that ICA component numbers, as stored in the variable ``picks``, were being used to index into a variable ``label_props`` that describes the style in which lines should be plotted. ``label_props`` is as long as ``picks``, not as long as the total number of ica components. Hence, the index of the component in ``picks``, not the ICA component number itself, should be used.

### CODE REVIEW

**Comment by jCalderTravis** [comment]
The documentation check failed (in the details it just says "block-unregistered-user"), but I could build the documentation on my own computer, with warnings.

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -54,6 +54,7 @@ Bugs
 - Fix bug where :ref:`mne show_fiff` could fail with an ambiguous error if the file is corrupt (:gh:`11800` by `Eric Larson`_)
 - Fix bug where annotation FIF files lacked an end block tag (:gh:`11800` by `Eric Larson`_)
 - Fix display of :class:`~mne.Annotations` in `mne.preprocessing.ICA.plot_sources` when the ``raw`` has ``raw.first_samp != 0`` and doesn't have a measurement date (:gh:`11766` by `Mathieu Scheltienne`_)
+- Fix bug that raised error when using `mne.preprocessing.ICA.plot_sources` under certain conditions (:gh:`11808` by `Joshua Calder-Travis`_)
```
Comment (drammock) [inline doc/changes/latest.inc:57]:
```suggestion
- Fix bug that used wrong indices for line/label styles (sometimes causing an ``IndexError``) in :meth:`mne.preprocessing.ICA.plot_sources` under certain conditions (:gh:`11808` by `Joshua Calder-Travis`_)
```

File: mne/viz/ica.py [hunk 2]
```diff
@@ -824,14 +824,14 @@ def _plot_ica_sources_evoked(evoked, picks, exclude, title, show, ica, labels=No
                 style = cat_styles[label_name]
                 label_props[label_idx] = (color, style)
 
-    for exc_label, ii in zip(exclude_labels, picks):
-        color, style = label_props[ii]
+    for pick_idx, (exc_label, i_pick) in enumerate(zip(exclude_labels, picks)):
```
Comment (drammock) [inline mne/viz/ica.py:827]: I think `pick_idx` and `i_pick` are too similar / confusing.  If I'm reading the code correctly, I think changing `i_pick` to just `pick` is the right repair. WDYT?

**Comment by drammock** [comment]
@jCalderTravis could you also add a test case (that fails on main and passes here)?  If it's just a few lines you could add it to this test instead of defining a whole new one:

https://github.com/mne-tools/mne-python/blob/9a2758d708f23e4e67a1d0205fe645a6d9f77aea/mne/viz/tests/test_ica.py#L387

Also add to the test a comment with the URL of this PR in it, so future devs know why those lines are there / what they're testing for.

File: mne/viz/ica.py (hunk 2 above)
Comment (jCalderTravis) [reply mne/viz/ica.py]: That sounds much better. Thanks for the suggestion, which I committed.

**Comment by jCalderTravis** [comment]
@drammock Thanks for the information about CircleCI. I added to the existing test as you suggested, and checked that it passes here but fails on main. Let me know if there is anything else you would like me to change.

File: mne/viz/tests/test_ica.py [hunk 3]
```diff
@@ -408,6 +408,13 @@ def test_plot_ica_overlay():
     pytest.raises(TypeError, ica.plot_overlay, raw[:2, :3][0])
     pytest.raises(TypeError, ica.plot_overlay, raw, exclude=2)
     ica.plot_overlay(raw)
+
+    # Check that the issue addressed by the following pull request is fixed:
+    # https://github.com/mne-tools/mne-python/pull/11808
+    # Issue meant that an error would be raised by plot_sources, unless picks
+    # were consecutive, starting at 0.
+    ica.plot_sources(eog_epochs.average(), picks=1)
```
Comment (drammock) [inline mne/viz/tests/test_ica.py:416]: oops, I got confused and guided you to the wrong place!  This is the `plot_overlay` test, we should add your new lines to the `plot_sources` test:

https://github.com/mne-tools/mne-python/blob/1704513f46f2596f7dccbb520e30804324999409/mne/viz/tests/test_ica.py#L266

Also a shorter comment would be ok, something like

```suggestion
    # regression test for `IndexError` when passing non-consecutive picks or consecutive
    # picks not including `0` (https://github.com/mne-tools/mne-python/pull/11808)
    ica.plot_sources(eog_epochs.average(), picks=1)
```

Comment (jCalderTravis) [reply mne/viz/tests/test_ica.py]: Ahh, sorry I didn't notice that. I moved the test to the correct place. Thanks for editing the comment.

**Comment by jCalderTravis** [comment]
I noticed the docs tests passed last time after you restarted them. And I have not edited the docs, so I think they should simply pass again this time.

**Review by drammock — APPROVED** [review]

**Comment by larsoner** [comment]
Thanks @jCalderTravis !

**Pull request merged**